config/initializers/aws.rb

```diff
@@ -1,15 +1,16 @@
 CarrierWave.configure do |config|
-  config.fog_provider = 'fog/aws'                     # required
-  config.fog_credentials = {
-    provider:              'AWS',                        # required
-    aws_access_key_id:     ENV['AWS_ACCESS_KEY'],            # required
-    aws_secret_access_key: ENV['AWS_SECRET_KEY'],                        # required
-    region:                'us-west-2',
-    path_style:            true            
-    # host:                  's3.example.com',             # optional, defaults to nil
-    # endpoint:              'https://s3.example.com:8080' # optional, defaults to nil
-  }
-  config.fog_directory  = 'servicealliance'                          # required
-  config.fog_public     = false                                        # optional, defaults to true
-  config.fog_attributes = { 'Cache-Control' => "max-age=#{365.day.to_i}" } # optional, defaults to {}
+  if Rails.env.production?
```

Comment:
I agree with you about environment conditionals. But i don't think you can avoid them completely in project, seems a bit extreme. I generally don't mind id enviroment conditionals are in the config/initializers/ folder, since those just alter application setup and doesn't alter code behavior.

I'm not really turning off CarrierWave here,  I just don't set any AWS related configuration here. It uses local hard drive to store files and avoids calls to AWS completely (I usually code without internet or with mobile tethering - having some code calling external service is a productivity killer for me)